fw/http_msg.c

```diff
@@ -652,17 +652,16 @@ __tfw_http_msg_add_str_data(TfwHttpMsg *hm, TfwStr *str, void *data,
 
 	if (TFW_STR_EMPTY(str)) {
 		if (!str->data)
-			__tfw_http_msg_set_str_data(str, data, skb);
+			__tfw_str_set_data(str, data, skb);
```

Comment:
Before this goes further, let me bring up this thought.
The change seems to break the naming schema to a certain degree. `str` functions are just that - they work on strings in their `TFW` implementation. These functions are intentionally written in the way that avoids direct dealing with `skb`. Now, `skb` appear whenever there's a HTTP message, and skb carries data, hence the appropriate naming for functions that deal with strings that are set on data in skb. Granted, the current naming may be too heavy and excessive (meaning - too long names), but they reflect the idea correctly.
New naming breaks this convention. What's more, it brings `skb` to `str.c` where all functions are _pure_ so far.
So, perhaps, there's a better way? Perhaps, `tfw_strcpy_prep()` doesn't belong there?
Hopefully I am not way off with this rant. Thanks.

Reply:
Though i do kind of agree with you, i must admit that `tfw_strcpy_prep()` still belongs here IMO. Yes, it sets `skb` explicitly, but other than that it's not that different from the other `tfw_str*` functions. It deals with chunks and other `TfwStr` members just like the others. `skb` is a legit member of `TfwStr` and no additional `sk_buff` manipulations have been introduced except for explicit assignment.
@krizhanovsky, @const-t what do you guys think?

Reply:
`tfw_strcpy_prep` in my opinion not a good name, because it can be used not only for preparing structure for `tfw_strcpy`, but it suitable to reside in `str.c`. `tfw_strcpy` almost copy of `tfw_strdup` except copy and `__tfw_str_set_data` call. `__tfw_str_set_data` can be replaced with explicit assign of skb or something like this `*dc = *data_str;`. Maybme I am wrong, but `__tfw_str_set_data` really can misled, name indicates we can set data only with skb.